integration-tests/client/test/core.test.ts

```diff
@@ -0,0 +1,355 @@
+            coin: `${transferAmount}${UATOM}`,
+            sender: gaiajs.address,
+            receiver: stridejs.address,
+        });
+
+        // Wait a bit for transfers to complete
+        await sleep(5000);
+
+        // Get final balances
+        const { balance: { amount: strideFinalStrideBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: USTRD,
+            });
+
+        const { balance: { amount: strideFinalAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const gaiaFinalAtomBalance = await gaiajs.query.bank.balance(gaiajs.address, UATOM);
+
+        console.log("Final balances:");
+        console.log(`Stride USTRD: ${strideFinalStrideBalance}`);
+        console.log(`Stride ATOM: ${strideFinalAtomBalance}`);
+        console.log(`Gaia ATOM: ${gaiaFinalAtomBalance.amount}`);
+
+        // Calculate and verify balance changes
+        const strideBalanceDiff = BigInt(strideInitialStrideBalance) - BigInt(strideFinalStrideBalance);
+        const strideAtomBalanceDiff = BigInt(strideFinalAtomBalance) - BigInt(strideInitialAtomBalance);
+        const gaiaAtomBalanceDiff = BigInt(gaiaInitialAtomBalance.amount) - BigInt(gaiaFinalAtomBalance.amount);
+
+        console.log("Balance differences:");
+        console.log(`Stride USTRD diff: ${strideBalanceDiff}`);
+        console.log(`Stride ATOM diff: ${strideAtomBalanceDiff}`);
+        console.log(`Gaia ATOM diff: ${gaiaAtomBalanceDiff}`);
+
+        // Verify the transfers worked (accounting for gas fees)
+        // USTRD should decrease on Stride by at least the transfer amount (plus gas fees)
+        expect(strideBalanceDiff).toBeGreaterThanOrEqual(BigInt(transferAmount));
+        expect(strideBalanceDiff).toBeLessThan(BigInt(transferAmount + 1000000)); // increased gas fee limit
+
+        // ATOM should increase on Stride by exactly the transfer amount
+        expect(strideAtomBalanceDiff).toBe(BigInt(transferAmount));
+
+        // ATOM should decrease on Gaia by at least the transfer amount (plus gas fees)
+        expect(gaiaAtomBalanceDiff).toBeGreaterThanOrEqual(BigInt(transferAmount));
+        expect(gaiaAtomBalanceDiff).toBeLessThan(BigInt(transferAmount + 1000000)); // increased gas fee limit
+    }, 120_000);
+
+    test("Liquid Stake Mint and Transfer", async () => {
+        const stridejs = strideAccounts.user;
+        const gaiajs = gaiaAccounts.user;
+        const stakeAmount = 10000000;
+
+        // Get initial balances
+        const { balance: { amount: strideInitialAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
```

Comment:
this structure is repeated in a lot of places, might make sense to move it to a utility function! where you could instead do:

```
const strideInitialAtomBalance = await getBalance({client: stridejs, address: stridejs.address, denom:  ATOM_DENOM_ON_STRIDE})
```